plugin-api/src/main/java/org/hyperledger/besu/plugin/services/securitymodule/data/PublicKey.java

```diff
@@ -12,23 +12,17 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-package org.hyperledger.besu.crypto;
+package org.hyperledger.besu.plugin.services.securitymodule.data;
 
-import org.apache.tuweni.bytes.Bytes;
-
-public class PublicKey {
-
-  private final Bytes encoded;
+import org.hyperledger.besu.plugin.Unstable;
 
-  public PublicKey(final Bytes encoded) {
-    this.encoded = encoded;
-  }
-
-  public static PublicKey create(final Bytes encoded) {
-    return new PublicKey(encoded);
-  }
+import org.apache.tuweni.bytes.Bytes;
 
-  public Bytes getEncoded() {
-    return encoded;
-  }
+/**
+ * Represents a function which provides encoded Bytes of Public Key (such as SECP256K1 Public Key).
+ */
+@Unstable
+@FunctionalInterface
+public interface PublicKey {
+  Bytes getEncoded();
```

Comment:
unfortunately, this is kinda dangerous - when security classes offer a getEncoded() they _typically_ imply an ASN1 encoding - not just a set of raw bytes.
So if we offer this as a function on a public key, we may need to be more explicit about what we expect.
[Given this is part of the interface, it will require javadoc.]

Reply:
@rain-on @CjHare What is the encoding used by SECP256K1 implementation? org/hyperledger/besu/crypto/SECP256K1.java:176